From efba7a0422f272c31584829ff57b21303edb1100 Mon Sep 17 00:00:00 2001
From: Tom Stellard <tstellar@redhat.com>
Date: Mon, 29 May 2017 11:08:35 -0400
Subject: [PATCH] Driver: Don't mix system tools with devtoolset tools on RHEL

For example, we don't want to mix a devtoolset gcc with system ld, because
they don't always work together.
---
 lib/Driver/ToolChains.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/lib/Driver/ToolChains.cpp b/lib/Driver/ToolChains.cpp
index 5ddc32b..1dd34ae 100644
--- a/lib/Driver/ToolChains.cpp
+++ b/lib/Driver/ToolChains.cpp
@@ -4111,6 +4111,15 @@ Linux::Linux(const Driver &D, const llvm::Triple &Triple, const ArgList &Args)
 
   Distro Distro(D.getVFS());
 
+  if (Distro.IsRedhat()) {
+    // On RHEL, we want to add a bin directory that is relative to the detected
+    // gcc install, because if we are using devtoolset gcc then we want to
+    // use other tools from devtoolset (e.g. ld) instead of the standard system
+    // tools.
+    PPaths.push_back(Twine(GCCInstallation.getParentLibPath() +
+                     "/../bin").str());
+  }
+
   if (Distro.IsOpenSUSE() || Distro.IsUbuntu()) {
     ExtraOpts.push_back("-z");
     ExtraOpts.push_back("relro");
-- 
1.8.3.1

